lib/models.py

```diff
@@ -2,13 +2,267 @@
 import openai
 from typing import Any, Callable, List, Optional
 
-class CoT():
+import numpy as np
+import itertools
+
+class ToT():
+    """
+    Creates a ToT wrapper around an LLM. 
```

Comment:
In the comment, I'd spell out `tree of thought` so that people know what ToT is.